julia/test/MyGame/Example/Monster.jl

```diff
@@ -0,0 +1,83 @@
+# automatically generated by the FlatBuffers compiler, do not modify
+
+# module: Example
+
+import ..InParentNamespace
+import FlatBuffers
+
+FlatBuffers.@with_kw mutable struct Monster{T}
```

Comment:
This is nice and short, but.. as I understand it, this generates an "object API", where you construct an object which then gets serialized. The whole point of FlatBuffers design is that it can create (and in particular, read) serialized data without creating language objects first, but directly.

Reply:
I feel like this is a question for @quinnj. There is a lower-level API in Julia for reading/writing flatbuffers directly however it relies on metaprogramming to determine the size of struct fields, meaning you have to construct a language object in order to be able to serialize one. Just so I'm clear, is performance the reason you want to avoid creating language objects? If so, we could run some tests.

Reply:
Yes, it is entirely about performance. If Julia's meta programming / JIT can make this just as fast as an API that constructs serialized data directly, then that is great also.

Reply:
Okay here is my hokey performance benchmark.

```
julia> include("MyGame.jl")
julia> using FlatBuffers, .MyGame
julia> mutable struct HpTable
          hp::Int
       end
julia> struct HpStruct
          hp::Int
       end
julia> io = IOBuffer()
julia> @time [FlatBuffers.serialize(io, MyGame.Example.Monster(;hp=42)) for _ = 1:1e5];
 14.450891 seconds (16.88 M allocations: 2.280 GiB, 1.39% gc time)

julia> @time [FlatBuffers.serialize(io, HpTable(42)) for _ = 1:1e5];
  1.284482 seconds (3.55 M allocations: 220.096 MiB, 1.91% gc time)

julia> @time [FlatBuffers.serialize(io, HpStruct(42)) for _ = 1:1e5];
  0.497959 seconds (1.28 M allocations: 103.010 MiB, 4.31% gc time)

julia> @time [FlatBuffers.serialize(io, 42) for _ = 1:1e5];
  0.429897 seconds (1.26 M allocations: 79.441 MiB, 1.92% gc time)
```
So it seems constructing a whole object (at least in the case of a Monster) consumes an order of magnitude more time and memory than constructing a one-element table, which is about 3 times slower than constructing a one-element struct. Sounds like the ability to create serialised data directly would definitely be a big performance win. However, it would be a significant API breaking change to the FlatBuffers julia package (which is already in use by a non-zero amount of people). 

I'd like the addition of Julia to be true to the original intent of FlatBuffers, and for the generated code to be as performant as possible. I'm prepared to put in some effort to make this happen. It would be a significant increase in scope on the current changes, however. Could we consider integrating Julia support as-is (i.e. suboptimal), and later adding the API for direct serialization? @quinnj do you think this could be a "1.0" breaking release of the FlatBuffers package?

Reply:
We can certainly go forward w/ a better design here; even if it means making breaking changes. The path forward would be to implement the changes, and put package version limits on downstream packages; then to go the extra mile, do pull requests for downstream packages to switch to the new APIs and then they can do new releases requiring the new FlatBuffers version. Happy to help navigate all that.

Reply:
We are only just adding official support for Julia to FlatBuffers, so I'd say breaking the API should not be a concern, doing the right thing is more important now. Those existing users should probably continue to use the existing code and the upgrade at their convenience.